features/networking/sriov.feature

```diff
@@ -69,3 +71,789 @@ Feature: Sriov related scenarios
+    """
+    When I run the :delete admin command with:
+      | object_type       | net-attach-def        |
+      | object_name_or_id | intel-netdevice-rhcos |
+      | namespace         | <%= cb.usr_project1%> |
+    Then the step should succeed
+    And I wait up to 30 seconds for the steps to pass:
+    """
+    When I run the :get admin command with:
+      | resource  | net-attach-def        |
+      | namespace | <%= cb.usr_project1%> |
+    Then the step should succeed
+    And the output should contain "intel-netdevice-rhcos"
+    """
+
+  # @author zzhao@redhat.com
+  # @case_id OCP-25790
+  @admin  
+  @destructive
+  Scenario: SR-IOV network config daemon can be set by nodeselector
+    Given the sriov operator is running well
+    And all existing pods are ready with labels:
+      | app=sriov-network-config-daemon |
+    Given configDaemonNodeSelector set to false in sriovoperatorconfig
+    And I wait for the resource "pod" named "<%= pod.name %>" to disappear within 120 seconds
+    Given configDaemonNodeSelector set to true in sriovoperatorconfig
+    Then all existing pods are ready with labels:
+      | app=sriov-network-config-daemon |
+
+  # @author zzhao@redhat.com
+  # @case_id OCP-26134
+  @destructive
+  @admin
+  Scenario: sriov can be shown in Metrics and telemetry
+    Given the sriov operator is running well
+    Given I obtain test data file "networking/sriov/sriovnetworkpolicy/mlx277-netdevice.yaml"
+    Given I create sriov resource with following:
+       | cr_yaml       | mlx277-netdevice.yaml    |
+       | cr_name       | mlx277-netdevice         |
+       | resource_type | sriovnetworknodepolicies |
+    Then the step should succeed
+    And I wait up to 500 seconds for the steps to pass:
+    """
+    When I run the :get admin command with:
+      | resource    | sriovnetworknodestates           |
+      | namespace   | openshift-sriov-network-operator |
+      | o           | yaml                             |
+    Then the step should succeed
+    And the output should contain "mlx277-netdevice"
+    And the output should contain "syncStatus: Succeeded"
+    And the output should contain "vfID: 1"
+    """
+    Given I switch to the first user
+    And I have a project
+    And evaluation of `project.name` is stored in the :usr_project clipboard
+    Given I obtain test data file "networking/sriov/sriovnetwork/mlx277netdevice.yaml"
+    Given I create sriov resource with following:
+       | cr_yaml       | mlx277netdevice.yaml |
+       | cr_name       | mlx277-netdevice     |
+       | resource_type | sriovnetwork         |
+       | project       | <%= cb.usr_project%> |
+    Then the step should succeed
+
+    And I wait up to 30 seconds for the steps to pass:
+    """
+    When I run the :get admin command with:
+      | resource  | net-attach-def        |
+      | namespace | <%= cb.usr_project%>  |
+    Then the step should succeed
+    And the output should contain "mlx277-netdevice"
```

Comment:
Above 5 lines can be shorten as `And admin checks that the "mlx-xxx" network_attachment_definition exists in the "yyy" project`

Reply:
seems not working well 
```
    And admin checks that the "intel-netdevice-rhcos" network_attachment_definition exists in the "<%= cb.usr_project %>" project # features/step_definitions/resources.rb:66
      [12:55:30] INFO> Shell Commands: oc get networkattachmentdefinition.k8s.cni.cncf.io intel-netdevice-rhcos --output=yaml --kubeconfig=/home/zzhao/workdir/dhcp-140-240-zzhao/ocp4_admin.kubeconfig
      apiVersion: k8s.cni.cncf.io/v1
      kind: NetworkAttachmentDefinition
      metadata:
        annotations:
          k8s.v1.cni.cncf.io/resourceName: openshift.io/intelnetdevice
        creationTimestamp: "2020-09-24T12:55:30Z"
        generation: 1
        managedFields:
        - apiVersion: k8s.cni.cncf.io/v1
          fieldsType: FieldsV1
          fieldsV1:
            f:metadata:
              f:annotations:
                .: {}
                f:k8s.v1.cni.cncf.io/resourceName: {}
            f:spec:
              .: {}
              f:config: {}
          manager: sriov-network-operator
          operation: Update
          time: "2020-09-24T12:55:30Z"
        name: intel-netdevice-rhcos
        namespace: 1l51i
        resourceVersion: "4294821"
        selfLink: /apis/k8s.cni.cncf.io/v1/namespaces/1l51i/network-attachment-definitions/intel-netdevice-rhcos
        uid: 42317c78-874a-4897-b332-92e8415a17d2
      spec:
        config: '{ "cniVersion":"0.3.1", "name":"intel-netdevice-rhcos", "type":"sriov","vlan":0,"spoofchk":"on","trust":"off","vlanQoS":0,"ipam":{"type":"host-local","subnet":"10.56.217.0/24","rangeStart":"10.56.217.171","rangeEnd":"10.56.217.181","routes":[{"dst":"0.0.0.0/0"}],"gateway":"10.56.217.1"} }'
      
      [12:55:34] INFO> Exit Status: 0
      hash not from a NetworkAttachmentDefinition: expected  but was NetworkAttachmentDefinition (RuntimeError)
      /home/zzhao/myrepo/verification-tests/lib/openshift/resource.rb:140:in `update_from_api_object'
      /home/zzhao/myrepo/verification-tests/lib/openshift/resource.rb:82:in `get'
      /home/zzhao/myrepo/verification-tests/lib/openshift/resource.rb:43:in `exists?'
      /home/zzhao/myrepo/verification-tests/lib/openshift/resource.rb:56:in `get_checked'
      /home/zzhao/myrepo/verification-tests/features/step_definitions/resources.rb:71:in `/^(I|admin) checks? that the "(.+?)" (\w+) exists(?: in the(?: "(.+?)")? project)?$/'
      features/networking/sriov.feature:119:in `And admin checks that the "intel-netdevice-rhcos" network_attachment_definition exists in the "<%= cb.usr_project %>" project'
      #"""
      #And I use the "<%= cb.usr_project%>" project

```

Reply:
Fix in https://github.com/openshift/verification-tests/pull/1566